src/provider/controller/ClientEventController.ts

```diff
@@ -28,10 +29,10 @@ export class ClientEventController {
                     type: 'notification-created',
                     notification: mutable(notification)
                 };
-                eventPump.push<NotificationCreatedEvent>(source.uuid, event);
+                await eventPump.push<NotificationCreatedEvent>(source.uuid, event);
```

Comment:
Note that event pump is potentially very long running, since it can potentially be starting an application. Worried about the implications of awaiting this

Reply:
We decided this is fine, right?

Reply:
Yeah nothing is awaiting this currently.